src/openassetio-python/CMakeLists.txt

```diff
@@ -107,20 +107,47 @@ if (OPENASSETIO_ENABLE_TESTS AND Python_Interpreter_FOUND)
         DEPENDS "${installed_pip}"
     )
 
+    # One-time step to download test-specific dependencies
+    # so we don't always need an online connection for running ctest
+    set(test_deps_dir "${CMAKE_BINARY_DIR}/testDeps")
+    add_custom_command(
+        OUTPUT "${test_deps_dir}"
+        COMMAND
+        "${installed_pip}" "download" "--destination-directory" "${test_deps_dir}"
+            "-r" "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+        DEPENDS "${installed_pip}" "${PROJECT_SOURCE_DIR}/tests/requirements.txt"
+    )
+
+    # One-time step to download project-specific dependencies
+    set(project_deps_dir "${CMAKE_BINARY_DIR}/projDeps")
+    add_custom_command(
+        OUTPUT "${project_deps_dir}"
```

Comment:
[fix] My bad, the downside of using a directory is that the `clean` target fails (at least on Linux+Ninja)

From my IDE
```
/usr/local/bin/cmake --build /home/dave/workspace/cloud/assetapi/OpenAssetIO/cmake-build-debug --target clean
[1/1] /snap/clion/184/bin/ninja/linux/ninja  -t clean 
FAILED: clean 
/snap/clion/184/bin/ninja/linux/ninja  -t clean 
Cleaning... ninja: error: remove(testDeps): Directory not empty
ninja: error: remove(/home/dave/workspace/cloud/assetapi/OpenAssetIO/cmake-build-debug/testDeps): Directory not empty
ninja: error: remove(projDeps): Directory not empty
ninja: error: remove(/home/dave/workspace/cloud/assetapi/OpenAssetIO/cmake-build-debug/projDeps): Directory not empty
13 files.
ninja: build stopped: subcommand failed.
```

One way around it that works is to add `ADDITIONAL_CLEAN_FILES` to a _target_ (can't be added to the custom command), e.g.
```
set_target_properties(openassetio-python-py-install PROPERTIES ADDITIONAL_CLEAN_FILES ${project_deps_dir})
```

To do this neatly I think we would have to have "dummy" targets that wrap the custom command, e.g.
```
add_custom_target(
    openassetio-python-download-project-deps
    DEPENDS "${project_deps_dir}"
)
set_target_properties(
    openassetio-python-download-project-deps 
    PROPERTIES ADDITIONAL_CLEAN_FILES ${project_deps_dir})
```
then depend on that target where `${project_deps_dir}` is required. This actually matches what [the docs](https://cmake.org/cmake/help/v3.22/command/add_custom_command.html) say

> Do not list the output in more than one independent target that may build in parallel or the two instances of the rule may conflict (instead use the [add_custom_target()](https://cmake.org/cmake/help/latest/command/add_custom_target.html#command:add_custom_target) command to drive the command and make the other targets depend on that one)

At this point, maybe having a signal file isn't so bad - it would mean `clean` would only clean the signal file not the downloaded content, but perhaps that would be an advantage...?

I'll leave the approach up to you.

Reply:
A general thought: I think it would be better that running clean should not delete those cached dependencies, or otherwise you destroy the ability to test once you run it and you happen to be without Internet at the time. What do you think?

Reply:
Note: make clean already was problematic before this change I think. If you run it, it will delete the pip executable in dist/bin and the test fails.

Reply:
The pip install's need a --force-reinstall option or the output files are not regenerated correctly, I just found.

Reply:
Eek, good spot!

Reply:
Addressed in https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/309/commits/c169f8fec784011e7c838c9c618ba36867a817f4